pkg/redshift/routes/routes.go

```diff
@@ -1,80 +1,41 @@
 package routes
 
 import (
-	"encoding/json"
-	"io"
-	"io/ioutil"
 	"net/http"
 
-	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
+	"github.com/grafana/grafana-aws-sdk/pkg/sql/routes"
 	"github.com/grafana/redshift-datasource/pkg/redshift"
 	"github.com/grafana/sqlds/v2"
 )
 
-type ResourceHandler struct {
-	ds redshift.RedshiftDatasourceIface
+type RedshiftResourceHandler struct {
+	routes.ResourceHandler
+	redshift redshift.RedshiftDatasourceIface
 }
 
-func New(ds *redshift.RedshiftDatasource) *ResourceHandler {
-	return &ResourceHandler{ds: ds}
+func New(api redshift.RedshiftDatasourceIface) *RedshiftResourceHandler {
+	return &RedshiftResourceHandler{routes.ResourceHandler{API: api}, api}
 }
 
-func write(rw http.ResponseWriter, b []byte) {
-	_, err := rw.Write(b)
-	if err != nil {
-		log.DefaultLogger.Error(err.Error())
-	}
-}
-
-func parseBody(body io.ReadCloser) (sqlds.Options, error) {
-	reqBody := sqlds.Options{}
-	b, err := ioutil.ReadAll(body)
-	if err != nil {
-		return nil, err
-	}
-	err = json.Unmarshal(b, &reqBody)
-	if err != nil {
-		return nil, err
-	}
-	return reqBody, nil
-}
-
-func sendResponse(res interface{}, err error, rw http.ResponseWriter) {
-	if err != nil {
-		rw.WriteHeader(http.StatusBadRequest)
-		write(rw, []byte(err.Error()))
-		return
-	}
-	bytes, err := json.Marshal(res)
-	if err != nil {
-		log.DefaultLogger.Error(err.Error())
-		rw.WriteHeader(http.StatusInternalServerError)
-		write(rw, []byte(err.Error()))
-		return
-	}
-	rw.Header().Add("Content-Type", "application/json")
-	write(rw, bytes)
+func (r *RedshiftResourceHandler) secrets(rw http.ResponseWriter, req *http.Request) {
+	secrets, err := r.redshift.Secrets(req.Context(), sqlds.Options{})
+	routes.SendResources(rw, secrets, err)
```

Comment:
Feels like there's something off with this package name. This is the route - shouldn't the package be called `resourceutils`, `httputils` or something?

Reply:
mmh, I am not sure I follow, this is the defining the route `/secrets` that's why the package is called `routes` (that's the naming we are following lately).